File: quark/test/emit/simple-math/js/simple-math.js
var _qrt = require("datawire-quark-core");

function main() {
    var four = (2) + (2);
    _qrt.print(_qrt.toString(four));
    var half = Math.floor((1) / (2));
    _qrt.print(_qrt.toString(half));
    var num = 314.0;
    var den = 100.0;
    var pi = (num) / (den);
    _qrt.print(_qrt.toString(pi));
    var pie = 3.14;
    _qrt.print(_qrt.toString(pie));
    var n = Math.floor((-(100)) / (3));
    _qrt.print(_qrt.toString(n));
    var m = Math.floor((100) / (-(3)));
    _qrt.print(_qrt.toString(m));
    var l = _qrt.modulo((100), (3));
    _qrt.print(_qrt.toString(l));
    var k = _qrt.modulo((-(100)), (3));
    _qrt.print(_qrt.toString(k));
}
exports.main = main;

main();

File: quark/test/emit/simple-math/py/simple_math.py
import os, sys

_Map = dict
class _List(list):
    def __repr__(self):
        return "[%s]" % ", ".join([str(e) for e in self])

def _println(obj):
    if obj is None:
        sys.stdout.write("null\n")
    else:
        sys.stdout.write("%s\n" % obj)


def main():
    four = (2) + (2);
    _println(str(four));
    half = (1) / (2);
    _println(str(half));
    num = 314;
    den = 100;
    pi = float(num) / float(den);
    _println(str(pi));
    pie = 3.14;
    _println(str(pie));
    n = (-(100)) / (3);
    _println(str(n));
    m = (100) / (-(3));
    _println(str(m));
    l = (100) % (3);
    _println(str(l));
    k = (-(100)) % (3);
    _println(str(k));


if __name__ == "__main__":
    main()
